smart-contracts/contracts/Unlock.sol

```diff
@@ -180,6 +260,39 @@ contract Unlock is
     return newLock;
   }
 
+  /**
+   * @dev Upgrade a Lock template implementation
+   * @param lockAddress the address of the lock to be upgraded
+   * @param version the version number of the template
+   */
+  function upgradeLock(address payable lockAddress, uint16 version) public returns(address) {
+    require(proxyAdminAddress != address(0), "proxyAdmin is not set");
+    require(lockAddress != address(0), "lockAddress can not be address 0");
+
+    // check perms
+    require(isLockManager(lockAddress, msg.sender) == true, "caller is not a manager of this lock");
+
+    // check version
+    IPublicLock lock = IPublicLock(lockAddress);
+    uint16 currentVersion = lock.publicLockVersion();
+    require( version == currentVersion + 1, 'version error: only +1 increments are allowed');
+
+    // make our upgrade
+    address impl = _publicLockImpls[version];
+    TransparentUpgradeableProxy proxy = TransparentUpgradeableProxy(lockAddress);
+    proxyAdmin.upgrade(proxy, impl);
```

Comment:
I assume this will fail for "old" (non upgradable) locks? It would be good to actually have a way to find "upgradable locks" so the UI only shows an upgrade button when the lock is indeed upgradable? (or do we just say any lock prior to 10 is not upgradable?)

Reply:
yes the >10 solution seems the most straightforward

Reply:
On the FE only then. so we save gas :)